FirebaseDatabase/Tests/Integration/FIRDatabaseQueryTests.m

```diff
@@ -4468,4 +4468,67 @@ - (void)testGetSkipsPersistenceCacheWhenOnline {
   WAIT_FOR(done);
 }
 
+- (void)testGetTriggersListener {
+  FIRDatabase* db = [self databaseForURL:self.databaseURL name:@"fir-multi-shards"];
+
+  FIRDatabaseReference* ref = [db referenceWithPath:@"james"];
+  FIRDatabaseReference* children = [ref child:@"children"];
+
+  __block BOOL done = NO;
+
+  [children removeValueWithCompletionBlock:^(NSError* _Nullable error,
+                                             FIRDatabaseReference* _Nonnull ref) {
+    XCTAssertNil(error);
+    done = YES;
+  }];
+
+  WAIT_FOR(done);
+  done = NO;
+
+  [[children childByAutoId] setValue:@{@"name" : @1}
```

Comment:
One difference between this test and the issue is that the test does not write values from the same instance, but rather all values exist on the backend. We can emulate this using a reader/writer test setup.